### DISCUSSION TITLE
CString::into_raw() memory leaks

### DISCUSSION THREAD

**Discussion by jgriffitts** [fltk-rs/fltk-rs#1482] [General]
While playing with the new wrapper for `Fl_Terminal` widget, I noticed that memory usage was slowly growing. After a little investigation, I found that the `Append()` method calls were using `CString::into_raw()` to create CStrings to pass to the cfltk interface, but never recovering and dropping those CStrings afterward. [(ref)](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.into_raw).

I just submitted a PR to fix this in Terminal and SimpleTerminal (for what that's worth) but I see a lot of other uses of `into_raw()` that make me worry. Some of them are documented in fltk as creating data structures that will never be freed, but others may represent slow memory leaks. In most cases, they're in methods that will not be called very often so the memory leak would be minimal, but I'm wondering if we should try to fix them anyway. Probably not a priority.

**Comment by MoAlyousef** [comment]
Hi @jgriffitts 
It's a complex issue. Sometimes FLTK requires a long lived string and sometimes it doesn't. It's sometimes documented like for widget constructor labels and sometimes not. I tried now and it seems to have changed, but the SimpleTerminal's append method's used to crash or print jibberish if the CString was passed as_ptr().
